### PR TITLE
update golang docs on CRD scoping

### PR DESCRIPTION
**Pull request by jberkhahn**
**Description of the change:**
Update the docs to discuss the various ways golang operators can control the scope of their CRD

**Motivation for the change:**
see issue #3684

### CODE REVIEW

File: website/content/en/docs/building-operators/golang/crds-scope.md [hunk 1]
```diff
@@ -6,29 +6,61 @@ weight: 60
 
 ## Overview
 
-The CustomResourceDefinition (CRD) scope can also be changed for cluster-scoped operators so that there is only a single 
-instance (for a given name) of the Custom Resource (CR) to manage across the cluster.
+Custom Resource Definitions (CRDs) contain a scope field that determines whether the resulting Custom Resource (CR)
+is cluster or namespace scoped. Operators can be build making use of this feature that then operator either on the entire cluster,
```
Comment (jmccormick2001) [inline website/content/en/docs/building-operators/golang/crds-scope.md:10]: Operators can be built using this feature to ...

Comment (jberkhahn) [reply website/content/en/docs/building-operators/golang/crds-scope.md]: wow, i don't know what happened there. thanks for the catch.

**Review by estroz — CHANGES_REQUESTED** [review]
A bunch of suggestions, the main themes being:
1. To clarify that operator scope != CRD scope
2. Go vs. non-Go configuration

File: website/content/en/docs/building-operators/golang/crds-scope.md [hunk 2]
```diff
@@ -6,29 +6,61 @@ weight: 60
...
 ## Overview
 
-The CustomResourceDefinition (CRD) scope can also be changed for cluster-scoped operators so that there is only a single 
-instance (for a given name) of the Custom Resource (CR) to manage across the cluster.
+Custom Resource Definitions (CRDs) contain a scope field that determines whether the resulting Custom Resource (CR)
+is cluster or namespace scoped. Operators can be built using this feature that then are available either on the entire cluster,
+or only in one or more namespaces. This might be done for several reasons: an operator author might use a namespaced-scoped operator
+to restrict access to a CR to a single namespace, or  to have different versions of similar CRs accessible in different namespaces.
+Alternatively, an operator author might want a cluster-scoped operator so all users can see and use a single operator.:
+This page details the various methods to control the scope of an operator.
```
[the remark this answers is not in the document]
Comment (jberkhahn) [reply website/content/en/docs/building-operators/golang/crds-scope.md:14]: uh, did you actually add the link here somewhere? i don't see it

File: website/content/en/docs/building-operators/golang/crds-scope.md [hunk 3]
```diff
@@ -6,29 +6,61 @@ weight: 60
...
 
-**NOTE**: When a `Manager` instance is created in the `main.go` file, it receives the namespace(s) as Options. 
+When creating a new API, the `--namespaced` flag controls whether the resulting Custom Resource will be cluster or namespace scoped.
+By default, `--namespaced` is set to `true`. An example command to create a cluster-scoped API would be:
+
+```bash
+$ operator-sdk create api --group cache --version v1alpha1 --kind Memcached --resource=true --controller=true --namespaced=false
+```
+
+## Set Scope Marker in types.go
```
[the remark this answers is not in the document]
Comment (jberkhahn) [reply website/content/en/docs/building-operators/golang/crds-scope.md:30]: isn't this page (building-operators/golang/crds-scope.md) only about golang operators?

File: website/content/en/docs/building-operators/golang/crds-scope.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (estroz) [reply website/content/en/docs/building-operators/golang/crds-scope.md:14]: `/docs/building-operators/golang/operator-scope` [already exists](https://sdk.operatorframework.io/docs/building-operators/golang/operator-scope/).

File: website/content/en/docs/building-operators/golang/crds-scope.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (estroz) [reply website/content/en/docs/building-operators/golang/crds-scope.md:30]: You're right, so I'd say do one of two things:
1. Copy the doc to ansible/helm dirs and remove the go-specific stuff from there and remove the non-go-specific stuff from this doc
2. Hoist this doc up to some top-level reference doc, and link to it from short reference docs in each language's `building-operators` dir

I would say implement my suggestions as-is, and do one of the above in a follow-up (or I can, I don't mind).

[the remark this answers is not in the document]
Comment (jberkhahn) [reply website/content/en/docs/building-operators/golang/crds-scope.md:29]: think i'm going to go with #2, and create a new issue to consolidate operator-type-agnostic help in some docs in the building-operators base directory. I'll open another issue for that.

**Review by estroz — APPROVED** [review]
/lgtm

**Pull request merged**